include/s3select_oper.h

```diff
@@ -1092,6 +1099,11 @@ void multi_values::push_value(value *v)
   }
 }
 
+void multi_values::push_key_path(value* v)  
+{
+  key_path = v->get_key();
```

Comment:
general comment.
the `value` is a complex structure, it may also contain multiple values.
it covers the use case where a single-projection may return multiple values (star-operation)

upon processing JSON input, each value gets its key (in case it is possible, depends on sql-operation)